ProductControllerTests.cs

```diff
@@ -0,0 +1,112 @@
+using AutoFixture;
+using CoreServices.Controllers;
+using CoreServices.DTO;
+using CoreServices.Repository;
+using CoreServices.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System;
+using Xunit;
+
+namespace CoreServices.Tests
+{
+    public class ProductControllerTests
+    {
+        private readonly ProductController _productController;
+        private readonly ILogger _logger;
+        private readonly Mock<IRepository> _repository;
+        private readonly Mock<ProductService> _mockProductService;
+
+        public ProductControllerTests()
+        {
+            _logger = new Mock<ILogger>().Object;
+            _repository = new Mock<IRepository>();
+            _mockProductService = new Mock<ProductService>(_repository.Object);

```

Comment:
_repository = new Mock<IRepository>();
_repository is not needed.
 _mockProductService = new Mock<ProductService>(); --> This should be fine